src/solver.rs

```diff
@@ -315,6 +309,8 @@ impl<I: std::ops::Fn(Epoch, SV, usize) -> Option<InterpolationResult>> Solver<I>
             })
             .collect();
 
+        let earth_j2000 = self.almanac.frame_from_uid(EARTH_J2000).unwrap();
```

Comment:
The `Frame` type implement `Copy`, so we just grab it once from the Almanac and copy it as many times as needed.